django/db/migrations/loader.py

```
@@ -316,6 +319,34 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
 
+    def check_consistent_history(self, connection):
+        """
+        Checks that all migrations are applied after their dependencies. Raises
+        InconsistentMigrationHistory exception when migration in applied before
+        it's dependency
+        """
+        recorder = MigrationRecorder(connection)
+        applied = recorder.applied_migrations()
+        for migration in applied:
+            # If we don't know anything about this migration we skip it
+            if migration not in self.graph.nodes:
+                continue
+            # Check only parents from the same app
+            unapplied_parents = [
+                x.key for x in self.graph.node_map[migration].parents
+                if x not in applied and x[0] == migration[0]
+            ]
```

I haven't tried it out, but I'm pretty certian this only happens when you use `__first__`: https://github.com/Uran198/django/blob/ticket_25850/django/db/migrations/loader.py#L148 which is silently ignored when adding dependencies.